File: algorithms/S/Smallest Integer Divisible by K/Smallest Integer Divisible by K.py
class Solution:
    def smallestRepunitDivByK(self, k: int) -> int:
        if k % 2 == 0: return -1
        n = 1
        leng = 1
        mapp = {}
        while True:
            rem = n % k
            if rem == 0: return leng
            if rem in mapp : return -1
            mapp[rem] = True
            n = n*10 + 1
            leng += 1
        

File: algorithms/S/Smallest Integer Divisible by K/Smallest Integer Divisible by K.js
/**
 * @param {number} k
 * @return {number}
 */
var smallestRepunitDivByK = function(k) {
    if( 1 > k > 1e6 ) return -1;
    let val = 0;
    for (let i = 1; i < 1e6; i++) {
        val = (val * 10 + 1) % k;
        if (val === 0) return i;
    }
    return -1;
};
